src/main/java/ru/l1ttleO/boyfriend/Utils.java

```diff
@@ -55,16 +57,41 @@ public static String getBeep() {
         return out.toString();
     }
 
-    public static int getDurationMultiplied(final @NotNull String toParse) {
+    public static int getDurationMultiplied(final @NotNull String toParse) throws NumberFormatException {
         try {
+            return Integer.parseInt(toParse);
+        } catch (final @NotNull NumberFormatException ignored) {
+        }
+
+        int result = 0;
+        final Set<String> used = Collections.emptySet();
+        String[] buffer;
+        final StringBuilder input = new StringBuilder(toParse);
+        while (!input.isEmpty()) {
+            buffer = input.toString().split("\\d+$");
+            if (buffer.length < 2 || used.contains(buffer[1]))
+                throw new NumberFormatException();
+            used.add(buffer[1]);
+            input.setLength(input.length() - buffer[1].length());
+
             int multiplier = 1;
-            for (int i = 0; i < DURATION_TEXTS.length; i++) {
-                if (toParse.endsWith(DURATION_TEXTS[i][0])) return multiplier * Integer.parseInt(toParse.substring(0, toParse.length() - 1));
+            int i = 0;
+            for (; i < DURATION_TEXTS.length; i++) {
```

Comment:
```suggestion
            int i;
            for (i = 0; i < DURATION_TEXTS.length; i++) {
```
Reason: `'for' statement lacks initializer`

Reply:
what's the difference?

Reply:
looks less confusing i guess